### ISSUE TITLE
SearchBox: Example markup is off

### ISSUE THREAD

**Issue by aremes** [OfficeDev/Office-UI-Fabric#312]
if you switch the order of input & label, hover effects will work even if you hover over the label. Same goes for the click event which hides the label.. switch it around and the whole component is *much* nicer to use. simple fix :)

**Comment by wdo3650** [comment]
@aremes Thanks for the suggestion! We added the hovering behavior in code for more control over that state. For example if a CSS hover state were used, the blue background color would blink off and on when the clear button is clicked since the input field is no longer in a focused or hovered state. 

Let us know if you have another solution. We're always looking for valuable contributions from users such as yourself. If you like you could even submit a pull request to show us what you're thinking.

**Comment by aremes** [comment]
ahem.... wat? I know what and how you're adding behavior. That was not the point of the issue at all. I didn't even mention CSS?! wat?!

Let's try this once more.. a little slower:

Let's go to [your official demo site for the searchbox](http://dev.office.com/fabric/components/searchbox).

Move your mouse over the part of the box where it says "Search". Notice how nothing happens? The cursor doesn't change, nor is the input highlighted as you would expect when you're moving the mouse over that. Click onto the "Search" text. Try the same with the little magnifying-glass-icon. Notice how nothing happens? Doing that did not activate the field. I would certainly expect it to. Go ahead, and create a html input with "placeholder" attribute and [look at how that works](http://www.wufoo.com/html5/attributes/01-placeholder.html). So i'm not crazy.. this is what the user would *actually expect*

Now. let's look at the searchbox markup under /src/components/searchbox/searchbox.html

```
<div class="ms-SearchBox">
  <input class="ms-SearchBox-field">
  <label class="ms-SearchBox-label"><i class="ms-SearchBox-icon ms-Icon ms-Icon--search"></i>Search</label>
  <button class="ms-SearchBox-closeButton"><i class="ms-Icon ms-Icon--x"></i></button>
</div>
```

If I do *exactly what i said above*: "switch the order of input and label" so I end up with the following markup:

```
<div class="ms-SearchBox">
  <label class="ms-SearchBox-label"><i class="ms-SearchBox-icon ms-Icon ms-Icon--search"></i>Search</label>
  <input class="ms-SearchBox-field">
  <button class="ms-SearchBox-closeButton"><i class="ms-Icon ms-Icon--x"></i></button>
</div>
```

then all of a sudden, everything behaves as expected.

All this has been tested in Chrome  v48, IE v11 and FF v43 using the latest npm distribution of fabric.

Please let me know if you can reproduce this at all.

**Comment by wdo3650** [comment]
@aremes Thank you for taking the time to illustrate the behavior you are seeing in a detailed manner.  This behavior was noted in issue #266 and pull request #294. The fix for this bug should be on the master branch and a release is scheduled later this week that addresses this issue.

**Comment by Jahnp** [comment]
This has been fixed on master and will be included in the 3.0.0 release.
